### DISCUSSION TITLE
Competing Updates - Race Conditions

### DISCUSSION THREAD

**Discussion by maxwellmstein** [jamessimone/apex-rollup#272] [Q&A] [answered]
I have an application that dispatches multiple calls from an LWC based on user interaction. This can cause Child records to be updated sporadically/simultaneously (different children of the same parent. This is causing issues the MAX/MIN date rollups on the parent are incorrect. I think the solution would be to force the rollup to do a full query/recalculation of all the children each time. 

Has anyone encountered something like this? It would also potentially be interesting if there was a way to 'debounce' the logic so the roll ups ran X seconds/min after all the children stopped being edited.

**Comment by jamessimone** [comment, marked as answer]
@maxwellmstein I'm curious - do you have `Should Run Single Records Synchronously` checked off on the Org Default Rollup Control CMDT record? I _have_ had others report this issue, and this was the fix for datatables / records being updated through LWC's `updateRecord` API. Let me know!
